akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowFlattenMergeSpec.scala

```diff
@@ -209,5 +210,12 @@ class FlowFlattenMergeSpec extends StreamSpec {
       attributes.indexOf(Attributes.Name("inner")) < attributes.indexOf(Attributes.Name("outer")) should be(true)
     }
 
+    "work with flatMapConcat optimized GraphStages.SingleSource" in assertAllStagesStopped {
+      Source(0 to 3)
+        .flatMapConcat(elem ⇒ new GraphStages.SingleSource(elem))
```

Comment:
Does this mean we should make the constructor of `SingleSource` a public API or we keep it to ourselves?